### PR TITLE
OpenShift cluster setup in new third party solutions section

### PR DESCRIPTION
**Pull request by ludoo**

### CODE REVIEW

File: third-party-solutions/openshift/tf/bootstrap.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+data "google_storage_object_signed_url" "bootstrap-ignition" {
+  for_each    = toset(local.bootstrapping ? [""] : [])
+  bucket      = google_storage_bucket.bootstrap-ignition.name
+  path        = google_storage_bucket_object.bootstrap-ignition[""].name
+  credentials = file(local.fs_paths.credentials)
+}
+
+resource "google_compute_instance" "bootstrap" {
+  count        = local.bootstrapping ? 1 : 0
```
Comment (juliocc) [inline third-party-solutions/openshift/tf/bootstrap.tf:39]: In the previous two resources you used for_each for the conditional. Pick one for consistency

File: third-party-solutions/openshift/tf/bootstrap.tf [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      }
+    })
+    VmDnsSetting = "GlobalDefault"
+  }
+}
+
+resource "google_compute_instance_group" "bootstrap" {
+  project     = var.service_project.project_id
+  network     = data.google_compute_network.default.self_link
+  zone        = "${var.region}-${element(var.zones, 0)}"
```
Comment (juliocc) [inline third-party-solutions/openshift/tf/bootstrap.tf:82]: var.zones[0]

File: third-party-solutions/openshift/prepare.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+  return result
+
+
+def _check_convert_paths(**paths):
+  'Return dictionary of path objects, check they point to existing resources.'
+  logging.info('checking paths')
+  result = {}
+  for k, v in paths.items():
+    p = pathlib.Path(v).expanduser()
+    if not p.exists:
```
[the remark this answers is not in the document]
Comment (ludoo) [reply third-party-solutions/openshift/prepare.py]: it should, nice catch!

File: third-party-solutions/openshift/tf/bootstrap.tf (hunk 1 above)
Comment (ludoo) [reply third-party-solutions/openshift/tf/bootstrap.tf:39]: switched all to `count` for better legibility when referencing

File: third-party-solutions/openshift/tf/bootstrap.tf (hunk 2 above)
Comment (ludoo) [reply third-party-solutions/openshift/tf/bootstrap.tf:82]: gah....

Comment (rosmo) [reply third-party-solutions/openshift/tf/bootstrap.tf]: It's really a matter of personal preference, but I like doing `format("%s-%s", var.region, var.zones[0])` :)

Comment (ludoo) [reply third-party-solutions/openshift/tf/bootstrap.tf]: good point! I already submitted this, but feel free to send a PR (or I can do it if you prefer)!

**Pull request merged**